modules/dxp/apps/osb/osb-faro/osb-faro-contacts-demo/src/main/java/com/liferay/osb/faro/contacts/demo/internal/ContactsDemo.java

```diff
@@ -95,20 +118,195 @@ protected void deactivate() {
 		}
 	}
 
+	private void _createDemo() throws Exception {
+		_faroProject =
+			_faroProjectLocalService.fetchFaroProjectByCorpProjectUuid(
+				FaroPropsValues.FARO_PROJECT_ID);
+
+		if (_faroProject == null) {
+			_faroProject = _createFaroProject();
+		}
+
+		try {
+			_createUsers(_faroProject);
+		}
+		catch (Exception exception) {
+			_log.error("Unable to add users", exception);
+		}
+
+		if (_hasExistingData()) {
+			if (_log.isInfoEnabled()) {
+				_log.info("Skipping demo creation because of existing data");
+			}
+
+			return;
+		}
+
+		_naniteDemoCreatorService = new NaniteDemoCreatorService(
```

Comment:
You should initialize it in another method. We do not need to create these fields for every `_createDemo` call.